src/wallet/api/wallet.cpp

```diff
@@ -1429,75 +1428,91 @@ PendingTransaction *WalletImpl::createTransaction(const string &dst_addr, const
+                } else {
+                    r = tools::wallet2::parse_short_payment_id(payment_id, info.payment_id);
+                    if (r) {
+                        std::string extra_nonce;
+                        set_encrypted_payment_id_to_tx_extra_nonce(extra_nonce, info.payment_id);
+                        r = add_extra_nonce_to_tx_extra(extra, extra_nonce);
+                    }
                 }
-            }
 
-            if (!r) {
-                setStatusError(tr("payment id has invalid format, expected 16 or 64 character hex string: ") + payment_id);
-                break;
+                if (!r) {
+                    setStatusError(tr("payment id has invalid format, expected 16 or 64 character hex string: ") + payment_id);
+                    break;
+                }
             }
-        }
-        else if (info.has_payment_id) {
-            std::string extra_nonce;
-            set_encrypted_payment_id_to_tx_extra_nonce(extra_nonce, info.payment_id);
-            bool r = add_extra_nonce_to_tx_extra(extra, extra_nonce);
-            if (!r) {
-                setStatusError(tr("Failed to add short payment id: ") + epee::string_tools::pod_to_hex(info.payment_id));
-                break;
+            else if (info.has_payment_id) {
+                std::string extra_nonce;
+                set_encrypted_payment_id_to_tx_extra_nonce(extra_nonce, info.payment_id);
+                bool r = add_extra_nonce_to_tx_extra(extra, extra_nonce);
+                if (!r) {
+                    setStatusError(tr("Failed to add short payment id: ") + epee::string_tools::pod_to_hex(info.payment_id));
+                    break;
+                }
             }
-        }
-
-
-        //std::vector<tools::wallet2::pending_tx> ptx_vector;
-
-        try {
             if (amount) {
-                vector<cryptonote::tx_destination_entry> dsts;
                 cryptonote::tx_destination_entry de;
-                de.original = dst_addr;
+                de.original = dst_addr[i];
                 de.addr = info.address;
-                de.amount = *amount;
+                de.amount = (*amount)[i];
                 de.is_subaddress = info.is_subaddress;
                 de.is_integrated = info.has_payment_id;
+
+                if (payment_id_seen) {
+                    setStatusError(tr("a single transaction cannot use more than one payment id"));
+                    break;
+                }
+
                 dsts.push_back(de);
-                transaction->m_pending_tx = m_wallet->create_transactions_2(dsts, fake_outs_count, 0 /* unlock_time */,
-                                                                          adjusted_priority,
-                                                                          extra, subaddr_account, subaddr_indices);
+
+                if (info.has_payment_id) {
+                    payment_id_seen = true;
+                }
```

Comment:
This looks like it will trigger even when only one integrated address id is used. Did you try it and it worked ?

Reply:
Updated `payment_id_seen` logic, was wrong before.